C-Plus-Plus/search/Interpolation_Search.cpp

```diff
@@ -1,78 +1,102 @@
-		else
-			hi = pos - 1; 
-	} 
-	return -1; 
-} 
-
-
-int main()							 //Starting our main function 
-{ 
- 
-	int num,x,loc;          
-    std::cout << "Enter size of an array: ";
-    std::cin >> num;
-    int arr[num];						//Creating array of the size num
-    std::cout << "Enter elements: " << std::endl; 
-    
-    for (int i = 0; i < num; i++)              			//Enter the elements in the array
-    {
-        std::cin >> arr[i];
-    }
-    
-    //Sorting our array
-    
-	int n = sizeof(arr)/sizeof(arr[0]); 
-	sort(arr, arr+n); 
-  
-    cout << "\nArray after sorting using "
-         "default sort is : \n"<< std::endl; 
-    for (int i = 0; i < n; ++i)
-    {
-        cout << arr[i] << " " ;
-    
+int interpolationSearch(int arr[], int n, int x) {
+  int low = 0, high = (n - 1); //Assigning lower index as 0 and higher index as n-1
+
+  while (low <= high && x >= arr[low] && x <= arr[high]) {
+    if (low == high) {
+      if (arr[low] == x) return low;
+      return -1;
     }
-    cout<< std::endl;
-    
-    //Entering the searching element
-    std::cout << "Enter element to be searched: " << std::endl;
-	cin>>x;// Element to be searched 
-	int index = interpolationSearch(arr, n, x); 
-
-	// If element was found 
-	if (index != -1) 
-		cout << "Element found at index " << index;      //Returning the index pf the element
-	else
-		cout << "Element not found.";                    //Returning -1 as element has not found 
-		
-	return 0; 
-} 
+    // Probing the position with keeping uniform distribution in mind. 
+    int pos = low + (((double)(high - low) /
+      (arr[high] - arr[low])) * (x - arr[low]));
+
+    // Condition of element to be found 
+    if (arr[pos] == x)
+      return pos;
+
+    // If x is larger, x is in upper part 
+    if (arr[pos] < x)
+      low = pos + 1;
+
+    // If x is smaller, x is in the lower part 
+    else
+      high = pos - 1;
+  }
+  return -1;
+}
+
+int main() //Starting our main function 
+{
+
+  int num, x, loc;
+  std::cout << "Enter size of an array: ";
+  std::cin >> num;
+  int arr[num]; //Creating array of the size num
+  std::cout << "Enter elements: " << std::endl;
+
+  for (int i = 0; i < num; i++) //Enter the elements in the array
+  {
+    std::cin >> arr[i];
+  }
+
+  //Sorting our array
+
+  int n = sizeof(arr) / sizeof(arr[0]);
+  sort(arr, arr + n);
+
+  cout << "\nArray after sorting using "
+  "default sort is : \n" << std::endl;
+  for (int i = 0; i < n; ++i) {
+    cout << arr[i] << " ";
+
+  }
+  cout << std::endl;
+
+  //Entering the searching element
+  std::cout << "Enter element to be searched: " << std::endl;
+  cin >> x; //Element to be searched 
+  int index = interpolationSearch(arr, n, x);
+  //If element was found 
+  if (index != -1)
+    cout << "Element found at index " << index; //Returning the index pf the element
+  else
+    cout << "Element not found."; //Returning -1 as element has not found 
+
+  return 0;
+}
+
+
+//SAMPLE INPUT:
```

Comment:
@sharur7 update this to multi-line comment something like this
```
/*
...
*/
```

Reply:
Yes, I did the changes in both C++ and Python files.